File: modelo/paciente.py
from modelo.persona import Persona


class Paciente(Persona):
    def __init__(self, dni: str, nombre: str, edad: int, historia_clinica: str):
        super().__init__(dni, nombre, edad)

        if not isinstance(historia_clinica, str) or not historia_clinica.strip():
            raise ValueError("La historia clínica es obligatoria")

        self._historia_clinica = historia_clinica.strip()
        self._atenciones = []

    @property
    def rol(self) -> str:
        return "Paciente"

    @property
    def historia_clinica(self) -> str:
        return self._historia_clinica

    @property
    def atenciones(self) -> tuple:
        return tuple(self._atenciones)

    def agregar_atencion(self, atencion) -> None:
        self._atenciones.append(atencion)

    def mostrar_datos(self) -> str:
        return (
            super().mostrar_datos()
            + f"\nN° Historia Clínica: {self._historia_clinica}"
            + f"\nAtenciones registradas: {len(self._atenciones)}"
        )

File: src/centrosalud/Paciente.java
package centrosalud;

// Herencia: Paciente extiende Persona
public class Paciente extends Persona {
    // Composición: cada paciente tiene su propia historia clínica
    private HistoriaClinica historiaClinica;

    public Paciente(String dni, String nombres, String apellidos, String fechaNacimiento,
                     String numeroHistoria) {
        super(dni, nombres, apellidos, fechaNacimiento);
        this.historiaClinica = new HistoriaClinica(numeroHistoria); // se crea automáticamente
    }

    public HistoriaClinica getHistoriaClinica() {
        return historiaClinica;
    }

    public void solicitarCita() {
        System.out.println(getNombreCompleto() + " solicita una cita médica.");
    }

    // Delega en HistoriaClinica
    public void agregarAtencion(AtencionMedica atencion) {
        historiaClinica.agregarAtencion(atencion);
    }

    public void consultarHistoria() {
        historiaClinica.mostrarHistoria();
    }

    // Polimorfismo: sobrescribe mostrarDatos() de Persona
    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println("N° Historia Clínica: " + historiaClinica.getNumeroHistoria());
        System.out.println("Atenciones registradas: "
                + historiaClinica.getAtenciones().size());
    }
}